apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/spring/webmvc/DispatcherServletRenderInstrumentation.java

```diff
@@ -0,0 +1,93 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.spring.webmvc;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.impl.transaction.Span;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.annotation.Nullable;
+
+import static net.bytebuddy.matcher.ElementMatchers.declaresMethod;
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isInterface;
+import static net.bytebuddy.matcher.ElementMatchers.nameStartsWith;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public class DispatcherServletRenderInstrumentation extends ElasticApmInstrumentation {
+
+    private static final String SPAN_TYPE_DISPATCHER_SERVLET_RENDER = "template.dispatcher-servlet.render";
+    private static final String RENDER_METHOD = "Render";
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    private static void beforeExecute(@Advice.Argument(0) ModelAndView modelAndView,
+                                      @Advice.Local("span") Span span) {
+        if (tracer == null || tracer.activeSpan() == null)
+            return;
+        final AbstractSpan<?> parent = tracer.activeSpan();
+        span = parent
+            .createSpan()
+            .withType(SPAN_TYPE_DISPATCHER_SERVLET_RENDER)
+            .appendToName(RENDER_METHOD).appendToName(" ").appendToName(modelAndView.getViewName())
```

Comment:
As per javadoc, `modelAndView.getViewName()` can return `null`. That case should probably be handled. Either just use `Render` as the name then or maybe use `modelAndView.getView().getClass().getSimpleName()`. However, `getSimpleName()` allocates memory which is best to avoid by coping the contents after the last dot/dollar sign up to the end of the string to the name `StringBuilder`. Is it common that the `View` is implemented as an anonymous inner class? In that case, the simple class name would not be helpful.

So maybe as a first step, just add the view name unless `null`.